docs/csharp/language-reference/builtin-types/default-values.md

```diff
@@ -42,6 +44,9 @@ Console.WriteLine(n);  // output: (0, 0)
 
 At run time, if the <xref:System.Type?displayProperty=nameWithType> instance represents a value type, you can use the <xref:System.Activator.CreateInstance(System.Type)?displayProperty=nameWithType> method to invoke the parameterless constructor to obtain the default value of the type.
 
+> [!NOTE]
+> In C# 10.0 and later, a [structure type](struct.md) (which is a value type) may have an [explicit parameterless constructor](struct.md#parameterless-constructors-and-field-initializers) that may produce a non-default value of the type. Thus, we recommend using the `default` operator or the `default` literal to produce the default value of a type.
```

Comment:
@BillWagner Can I get into trouble even in earlier C# versions (let's say in C# 7) if I use the library written and compiled with C# 10?

Reply:
@pkulikov I don't think so. I suppose that if you were expecting the 0-bit pattern and didn't get it, that could be unexpected. But, I would think that would be part of the motivation for using a C# 10 library.

More important here, I think, would be to emphasize that the explicit parameterless constructor won't get called for arrays and stackalloc expressions.

Reply:
> More important here, I think, would be to emphasize that the explicit parameterless constructor won't get called for arrays and stackalloc expressions.

What makes it important in this article? This article is about the default values of C# types, and I'm afraid the content about parameterless constructor is starting to occupy too much space. We may say that the default values are used at array initialization, instead. What do you think?

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/struct#parameterless-constructors-and-field-initializers - already tells that explicit constructors are not called for arrays.

The article about `stackalloc` says that `The content of the newly allocated memory is undefined.` So, it's not even 0-bit. So, saying that explicit parameterless constructor is not called may confuse people into thinking that the default type value is used, which is not true.

Reply:
Thanks @pkulikov 

I like your analysis here. I'll approve, and :shipit: now